packages/client-proxy/main.go

```diff
@@ -164,4 +173,7 @@ func main() {
 		logger.Error("http service exited without error", zap.Int("port", port))
```

Comment:
I might be inclined to have this also be in a go routine (in the wait group.) 

Particularly since there are two services, I'd also be inclined to cancel the context in (at least) the error case (e.g. line 160). so that if one server crashes or exits before shutdown, you would want the rest of the service to shutdown.